admin/worker/billing_reporter.go

```diff
@@ -0,0 +1,247 @@
+package worker
```

Comment:
I think it's worth going over this file once more with an aim of simplifying the code where possible. There are some places with pretty deep nesting and if/else scenarios.

Also, have you considered syncing events in bulk for all orgs/projecst in one query (going over all events in the bucket) instead of querying and uploading per org/project? It seems the Orb APIs would support that and could reduce the latency and number of queries by a lot.

Reply:
Yeah its pretty busy. Tried simplifying a bit by breaking logic into functions to handle org and then its projects. Also changed reporting logic to first collect usage for all projects in an org and reporting all at once. Two more places to look into -
- Reporting usage to Orb for all orgs, not sure if we would hit issues with big payload 
- Usage availability is fetched for all projects of an org in one go from rill-cloud-metrics but fetching usage is still done per project per bucket per metric. Each project might have different buckets to report and metrics to report may change as per plan so creating a single api to fetch usage per plan can be hard to keep in sync.

Reply:
> Each project might have different buckets to report and metrics to report may change as per plan so creating a single api to fetch usage per plan can be hard to keep in sync.

My hope would be that we could simplify usage reporting a lot by not trying to do such a granular breakdown. It should also lead to fewer, more efficient queries to the DB. Unless Orb charges a lot of money for redundant metrics, it seems simpler to just ship all non-zero billable metrics to them and have the logic of which metrics to use implemented there.

Of course we would probably need to batch the uploads to Orb to not upload huge payloads.

I'm imagining a query like this on our side (pseudo-code):
```sql
SELECT instance_id, <date trunc> as event_time, metric_name, <agg> as value
FROM metrics
WHERE event_time > <from time> AND event_time < <to time>
GROUP BY ALL
ORDER BY 
```

Then in the worker, it just does a lookup and caches each `instance_id`, enriches the events with info about the instance's org, and does a bulk upload to Orb of e.g. every 1k events.